src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/RplFundingCalulationService.cs

```diff
@@ -0,0 +1,106 @@
+                bool? isDurationReducedByRpl,
+                DbSet<StandardFundingPeriod> standardFundingPeriods,
+                DbSet<FrameworkFundingPeriod> frameworkFundingPeriods
+            )
+        {
+            int? fundingBandMaximum = await GetFundingBandMaximum(courseCode, startDate, frameworkFundingPeriods, standardFundingPeriods);
+            decimal? percentageOfPriorLearning = CalculatePercentageOfPriorLearning(durationReducedByHours, trainingTotalHours);
+            decimal? minimumPercentageReduction = CalculateMinimumPercentageOfPriorLearning(percentageOfPriorLearning);
+            int? minimumPriceReduction = CalculateMinimumPriceReduction(fundingBandMaximum, minimumPercentageReduction);
+            bool rplPriceReductionError = HasRplPriceReductionError(trainingTotalHours, durationReducedByHours, priceReducedBy, isDurationReducedByRpl, minimumPriceReduction);
+
+            var rplFundingCalulation = new RplFundingCalulation
+            {
+                FundingBandMaximum = fundingBandMaximum,
+                PercentageOfPriorLearning = percentageOfPriorLearning,
+                MinimumPercentageReduction = minimumPercentageReduction,
+                MinimumPriceReduction = minimumPriceReduction,
+                RplPriceReductionError = rplPriceReductionError
+            };
+
+            return rplFundingCalulation;
+        }
+
+        private async Task<int?> GetFundingBandMaximum(string courseCode, DateTime? startDate, DbSet<FrameworkFundingPeriod> frameworkFundingPeriods, DbSet<StandardFundingPeriod> standardFundingPeriods)
+        {
+            if (string.IsNullOrEmpty(courseCode))
+                return null;
+
+            if (int.TryParse(courseCode, out var standardId))
+            {
+                var standard = await standardFundingPeriods.Where(c => c.Id.Equals(standardId)
+                    && c.EffectiveFrom <= startDate && (c.EffectiveTo == null || c.EffectiveTo >= startDate))
+                    .OrderByDescending(x => x.EffectiveFrom).FirstOrDefaultAsync();
+                return standard?.FundingCap;
+            }
+
+            var framework = await frameworkFundingPeriods.Where(c => c.Id.Equals(courseCode)
+                    && c.EffectiveFrom <= startDate && (c.EffectiveTo == null || c.EffectiveTo >= startDate))
+                .OrderByDescending(x => x.EffectiveFrom).FirstOrDefaultAsync();
+            return framework?.FundingCap;
+        }
+
+        private static decimal? CalculatePercentageOfPriorLearning(int? durationReducedByHours, int? trainingTotalHours)
+        {
+            if (durationReducedByHours == null || trainingTotalHours == null || trainingTotalHours == 0)
+                return null;
+            return (decimal)durationReducedByHours / trainingTotalHours * 100;
+        }
+
+        private static decimal? CalculateMinimumPercentageOfPriorLearning(decimal? percentageOfPriorLearning)
+        {
+            if (percentageOfPriorLearning == null)
```

Comment:
I don't think you need this null check - dividing a nullable decimal with a value of `null` results in `null` (but please check!).

Reply:
Yep, great find. Thanks. Fixed